File: ios/AtomicFileOps.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(AtomicFileOps, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(writeFile:(NSString)filePath withData:(NSString)data withOptions:(NSString)options
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/AtomicFileOps.swift
import Foundation

@objc(AtomicFileOps)
class AtomicFileOps: NSObject {
    @objc(writeFile:withData:withOptions:withResolver:withRejecter:)
    func writeFile(fileName: String, contents: String, encoding: String, resolve:@escaping RCTPromiseResolveBlock,reject:@escaping RCTPromiseRejectBlock) -> Void {
      AtomicFileHandler.writeFile(fileName: fileName, contents: contents, encoding: encoding) { (returnString, error) in
          guard error == nil else {
            reject("Error", error?.localizedDescription, error)
              return
          }
          resolve(returnString);
      }
    }
    
    @objc public static func requiresMainQueueSetup() -> Bool {
        return false
    }
}
